### PR TITLE
swiftlint: add linux support, 0.56.1 -> 0.56.2

### PR DESCRIPTION
**Pull request by ilaumjd**
## Description of changes

- Add Linux support.
- Update to the latest version.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/sw/swiftlint/package.nix [hunk 1]
```diff
@@ -36,7 +44,6 @@ stdenvNoCC.mkDerivation rec {
     license = licenses.mit;
     mainProgram = "swiftlint";
     maintainers = with maintainers; [ matteopacini ];
-    platforms = platforms.darwin;
```
Comment (Pandapip1) [inline pkgs/by-name/sw/swiftlint/package.nix:39]:
```suggestion
platforms = platforms.all;
```

**Comment by Pandapip1** [comment]
Result of `nixpkgs-review pr 338080` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>swiftlint</li>
  </ul>
</details>

**Review by Pandapip1 — CHANGES_REQUESTED** [review]
Missing dependencies

```
.cache/nixpkgs-review/pr-338080-1 via ❄️  impure (review-shell) 
❯ results/swiftlint/bin/swiftlint 
results/swiftlint/bin/swiftlint: /run/current-system/sw/share/nix-ld/lib/libcurl.so.4: no version information available (required by results/swiftlint/bin/swiftlint)
SourceKittenFramework/library_wrapper.swift:58: Fatal error: Loading libsourcekitdInProc.so failed
Illegal instruction (core dumped)
```

File: pkgs/by-name/sw/swiftlint/package.nix [hunk 2]
```diff
@@ -36,7 +44,7 @@ stdenvNoCC.mkDerivation rec {
...
+    platforms = platforms.all;
```
Comment (matteo-pacini) [inline pkgs/by-name/sw/swiftlint/package.nix:47]:
```suggestion
    platforms = platforms.darwin ++ platforms.linux;
```

`platforms.all` is too broad, and Swift may not even support those - I think it's best to restrict this to the actual platforms Swiftlint supports.

image

**Review by matteo-pacini — CHANGES_REQUESTED** [review]
Thanks for this! Left two comments.

Also, please check the other review comment here: https://github.com/NixOS/nixpkgs/pull/338080#pullrequestreview-2267651979

**Comment by ilaumjd** [comment]
> Missing dependencies
> 
> ```
> .cache/nixpkgs-review/pr-338080-1 via ❄️  impure (review-shell) 
> ❯ results/swiftlint/bin/swiftlint 
> results/swiftlint/bin/swiftlint: /run/current-system/sw/share/nix-ld/lib/libcurl.so.4: no version information available (required by results/swiftlint/bin/swiftlint)
> SourceKittenFramework/library_wrapper.swift:58: Fatal error: Loading libsourcekitdInProc.so failed
> Illegal instruction (core dumped)
> ```

Oh just checked that swiftlint requires swift 5.9 and nixpkgs only have 5.8 and I checked on my machine which I don't use swift from nixpkgs. I'm closing the PR since it won't be able to get it work.

**Comment by matteo-pacini** [comment]
@ilaumjd there was some work going on re Swift 5.9 - https://github.com/NixOS/nixpkgs/pull/256956 - but it might take some time before that happens.

Have you tried adding `curl` as build input and using `autoPatchElfHook` on the Linux binary?

**Comment by ilaumjd** [comment]
@matteo-pacini oh never heard about that, will check it later

**Pull request closed without merging**